sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -182,7 +208,9 @@ case class ConcatWs(children: Seq[Expression])
 
       ev.copy(
         code"""
-        $codes
+        boolean[] $isNullArgs = new boolean[${children.length - 1}];
+        Object[] $valueArgs = new Object[${children.length - 1}];
+        $argBuilds
```

Comment:
Don't we need to evaluate `evals.head.code`? Previously it is in `codes` but now `argBuilds` only includes `children.tail`.

Reply:
If I understand correctly, evals.head is delimiter, and it's used in `UTF8String.concatWs`.

Reply:
Ah OK I see the case, where delimiter is not literal. I'll add it. Thanks!